packages/form/src/components/CheckboxField/index.tsx

```diff
@@ -61,10 +62,11 @@ export const CheckboxField = <
       }}
       disabled={field.disabled}
       checked={!!field.value}
-      error={getError({ label: label ?? '' }, error)}
+      error={getError({ label: label ?? ariaLabel ?? name }, error)}
       ref={field.ref}
-    >
-      {children}
-    </Checkbox>
+      {...(label
+        ? { children: label, 'aria-label': undefined }
+        : { 'aria-label': ariaLabel as string })}
```

Comment:
if it can be undefined, no need to cast it as string, no ?

Reply:
apparently you need to or it thinks it might be undefined. I think types are quite broken when you use this method :/

Reply:
My bad, you made it mandatory on purpose, I didn't understand the use case before that, all good